pkg/controller/podautoscaler/replica_calculator.go

```diff
@@ -287,17 +279,23 @@ func (c *ReplicaCalculator) GetObjectPerPodMetricReplicas(statusReplicas int32,
 	return replicaCount, utilization, timestamp, nil
 }
 
-// @TODO(mattjmcnaughton) Many different functions in this module use variations
-// of this function. Make this function generic, so we don't repeat the same
-// logic in multiple places.
-func (c *ReplicaCalculator) getReadyPodsCount(namespace string, selector labels.Selector) (int64, error) {
+func (c *ReplicaCalculator) getNonemptyPodList(namespace string, selector labels.Selector) ([]*v1.Pod, error) {
```

Comment:
It's nice that you've made this method generic.  However the error message is too generic now.  If I see a log entry saying "unable to get pods while getting nonempty podList" that isn't going to tell me anything about what the controller was doing when it encountered an error.

Since this method is only used within `replica_calculator.go` (it's unexported), how about we keep the original error messages.  So the `getNonemptyPodList` method can optionally return an error saying "unable to get all pods while calculating replica count" or "no pods returned by selector while calculating replica count" (as before).  The call sites will be simpler but we won't lose any detail in the error messages.

Reply:
Already fixed.